### PR TITLE
Need at least one '/' after ':' for md auto links

### LINKED ISSUE #149475: `.md` file syntax issue: Follow link
Screenshot 2022-05-13 083513

### PR DESCRIPTION
**Pull request by MachineMitch21**
This PR fixes #149475

### CODE REVIEW

File: extensions/markdown-language-features/src/languageFeatures/documentLinkProvider.ts [hunk 1]
```diff
@@ -173,7 +173,7 @@ const referenceLinkPattern = /(^|[^\]\\])(?:(?:(\[((?:\\\]|[^\]])+)\]\[\s*?)([^\
 /**
  * Matches `<http://example.com>`
  */
-const autoLinkPattern = /\<(\w+:[^\>\s]+)\>/g;
+const autoLinkPattern = /\<(\w+:\/{1,}[^\>\s]+)\>/g;
```
Comment (jsoref) [inline extensions/markdown-language-features/src/languageFeatures/documentLinkProvider.ts:176]: Given that `+` is already used in this pattern...
```suggestion
const autoLinkPattern = /\<(\w+:\/+[^\>\s]+)\>/g;
```

**Review by mjbvz — CHANGES_REQUESTED** [review]
Thanks for looking into this but autolinks don't require a `/`. For example: `<mailto:user@example.com>`

**Comment by jsoref** [comment]
In theory one can register any arbitrary protocol `foo:` with their OS. Which means that yeah, it could be...

I think you could make the argument that: `<`+ [ `/` ] + xml-characters +`:` + xml-characters + [ space + attributes] + `>` shouldn't be treated as a protocol. It's really just an XML tag (or a closing tag).

It might make sense to use some zero-width assertions.

**Comment by mjbvz** [comment]
Thanks for looking into this

Based on these comments and how common mark handles the code, I think this is by-design. I've gone ahead and closed the original issue: https://github.com/microsoft/vscode/issues/149475#issuecomment-1129401447

**Pull request closed without merging**